src/segment-loader.js

```diff
@@ -761,18 +761,30 @@ export default class SegmentLoader extends videojs.EventTarget {
                              timeUntilRebuffer,
                              this.pendingSegment_.duration);
 
+    let minimumTimeSaving = 0.5
+
+    // If we are already rebuffering, increase the amount of variance we add to the
+    // potential round trip time of the new request so that we are not too agressive
+    // with switching to a playlst that might save us a fraction of a second.
+    if (timeUntilRebuffer <= TIME_FUDGE_FACTOR) {
+      minimumTimeSaving = 1;
+    }
+
     // Finally, don't abort early and trigger the downswitch if the playlist
```

Comment:
I don't think this comment (or the others in the conditional) add anything to the code itself.